### PR TITLE
Fixed issues around empty lobs in MySQL. Closes #1919

### LINKED ISSUE #1919: Centaur: empty_string test fails on MySQL databases
The following stack trace appears:

```
2017-01-27 11:43:10,369 cromwell-system-akka.dispatchers.engine-dispatcher-19 ERROR - WorkflowManagerActor Workflow 268771d3-0303-45b1-ba0f-9c93c27b6784 failed (during ExecutingWorkflowState): JobStore write failure: Invalid position in SerialClob object set
java.lang.Exception: JobStore write failure: Invalid position in SerialClob object set
	at cromwell.engine.workflow.lifecycle.execution.EngineJobExecutionActor$$anonfun$11.applyOrElse(EngineJobExecutionActor.scala:239)
	at cromwell.engine.workflow.lifecycle.execution.EngineJobExecutionActor$$anonfun$11.applyOrElse(EngineJobExecutionActor.scala:235)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
	at akka.actor.FSM$class.processEvent(FSM.scala:663)
	at cromwell.engine.workflow.lifecycle.execution.EngineJobExecutionActor.akka$actor$LoggingFSM$$super$processEvent(EngineJobExecutionActor.scala:29)
	at akka.actor.LoggingFSM$class.processEvent(FSM.scala:799)
	at cromwell.engine.workflow.lifecycle.execution.EngineJobExecutionActor.processEvent(EngineJobExecutionActor.scala:29)
	at akka.actor.FSM$class.akka$actor$FSM$$processMsg(FSM.scala:657)
	at akka.actor.FSM$$anonfun$receive$1.applyOrElse(FSM.scala:651)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:496)
	at cromwell.engine.workflow.lifecycle.execution.EngineJobExecutionActor.aroundReceive(EngineJobExecutionActor.scala:29)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
	at akka.actor.ActorCell.invoke(ActorCell.scala:495)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: javax.sql.rowset.serial.SerialException: Invalid position in SerialClob object set
	at javax.sql.rowset.serial.SerialClob.getSubString(SerialClob.java:269)
	at com.mysql.jdbc.PreparedStatement.setClob(PreparedStatement.java:3171)
	at com.zaxxer.hikari.proxy.PreparedStatementJavassistProxy.setClob(PreparedStatementJavassistProxy.java)
	at slick.driver.JdbcTypesComponent$JdbcTypes$ClobJdbcType.setValue(JdbcTypesComponent.scala:148)
	at slick.driver.JdbcTypesComponent$JdbcTypes$ClobJdbcType.setValue(JdbcTypesComponent.scala:146)
	at slick.jdbc.BaseResultConverter.set(JdbcResultConverter.scala:18)
	at slick.jdbc.BaseResultConverter.set(JdbcResultConverter.scala:10)
	at slick.relational.ProductResultConverter.set(ResultConverter.scala:71)
	at slick.relational.ProductResultConverter.set(ResultConverter.scala:46)
	at slick.relational.TypeMappingResultConverter.set(ResultConverter.scala:138)
	at slick.driver.JdbcActionComponent$InsertActionComposerImpl$MultiInsertAction$$anonfun$run$8$$anonfun$apply$1.apply(JdbcActionComponent.scala:519)
	at slick.driver.JdbcActionComponent$InsertActionComposerImpl$MultiInsertAction$$anonfun$run$8$$anonfun$apply$1.apply(JdbcActionComponent.scala:517)
	at slick.jdbc.JdbcBackend$SessionDef$class.withPreparedInsertStatement(JdbcBackend.scala:354)
	at slick.jdbc.JdbcBackend$BaseSession.withPreparedInsertStatement(JdbcBackend.scala:407)
	at slick.driver.JdbcActionComponent$ReturningInsertActionComposerImpl.preparedInsert(JdbcActionComponent.scala:636)
	at slick.driver.JdbcActionComponent$InsertActionComposerImpl$MultiInsertAction$$anonfun$run$8.apply(JdbcActionComponent.scala:517)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.immutable.List.map(List.scala:285)
	at slick.driver.JdbcActionComponent$InsertActionComposerImpl$MultiInsertAction.run(JdbcActionComponent.scala:522)
	at slick.driver.JdbcActionComponent$SimpleJdbcDriverAction.run(JdbcActionComponent.scala:32)
	at slick.driver.JdbcActionComponent$SimpleJdbcDriverAction.run(JdbcActionComponent.scala:29)
	at slick.backend.DatabaseComponent$DatabaseDef$$anon$2.liftedTree1$1(DatabaseComponent.scala:237)
	at slick.backend.DatabaseComponent$DatabaseDef$$anon$2.run(DatabaseComponent.scala:237)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

### PR DESCRIPTION
**Pull request by kshakir**
Only store null instead of "" for a clob.
Only store null instead of Array.empty[Byte] for a blob.
All lob columns are now nullable.
Added test to ensure that all lob columns are nullable.
Updated SqlConverters for converting to/from Option and empty lobs.

### CODE REVIEW

File: database/sql/src/main/scala/cromwell/database/sql/SqlConverters.scala [hunk 1]
```diff
@@ -18,21 +18,37 @@ object SqlConverters {
...
+  }
+
+  implicit class StringOptionToClobOption(val strOption: Option[String]) extends AnyVal {
+    def toClob: Option[Clob] = strOption.flatMap(_.toClob)
   }
 
-  implicit class StringToClob(val str: String) extends AnyVal {
-    def toClob: Clob = new SerialClob(str.toCharArray)
+  implicit class StringToClobOption(val str: String) extends AnyVal {
+    def toClob: Option[Clob] = if (str == "") None else Option(new SerialClob(str.toCharArray))
```
Comment (Horneth) [inline database/sql/src/main/scala/cromwell/database/sql/SqlConverters.scala]: `str.isEmpty` ?

**Comment by Horneth** [comment]
:+1:

File: database/sql/src/main/scala/cromwell/database/sql/SqlConverters.scala [hunk 2]
```diff
@@ -18,21 +18,37 @@ object SqlConverters {
...
-  implicit class BlobToBytes(val blob: Blob) extends AnyVal {
-    def toBytes: Array[Byte] = blob.getBytes(1, blob.length.toInt) // yes, it starts at 1
+  implicit class BytesOptionToBlob(val bytesOption: Option[Array[Byte]]) extends AnyVal {
+    def toBlob: Option[Blob] = bytesOption.flatMap(_.toBlob)
   }
 
-  implicit class StringToBlob(val bytes: Array[Byte]) extends AnyVal {
-    def toBlob: Blob = new SerialBlob(bytes)
+  implicit class BytesToBlob(val bytes: Array[Byte]) extends AnyVal {
+    def toBlob: Option[Blob] = if (bytes.length == 0) None else Option(new SerialBlob(bytes))
```
Comment (mcovarr) [inline database/sql/src/main/scala/cromwell/database/sql/SqlConverters.scala]: `bytes.isEmpty`

**Comment by mcovarr** [comment]
:+1:

**Pull request merged**